File: LC_977.java
class Solution {
    public int[] sortedSquares(int[] A) {
        
        int[] result = new int[A.length];
        
        int left = 0, right = A.length - 1, pos = A.length - 1;
        
        while (pos >= 0) {
            int leftCandidate = A[left] * A[left];
            int rightCandidate = A[right] * A[right];
            
            if (leftCandidate > rightCandidate) {
                result[pos] = leftCandidate;
                left++;
            }
            else {
                result[pos] = rightCandidate;
                right--;
            }
            pos--;
        }
        
        return result;
    }
}

File: LC_977.py
# 977. Squares of a Sorted Array

class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        result = [0] * len(nums)
        
        i, j = 0, len(nums) - 1
        
        for k in range(len(nums) - 1, -1, -1):
            if abs(nums[i]) > abs(nums[j]):
                result[k] = nums[i] ** 2
                i += 1
            else:
                result[k] = nums[j] ** 2
                j -= 1
        
        return result
    